### ISSUE TITLE
Encryption of secret values for GitOps

### ISSUE THREAD

**Issue by mumoshu** [Aptomi/aptomi#249]
So that I could safely git-commit e.g. helm's values.yaml contains sensitive data, which could be decrypted when I run aptomictl.

Would it be within the scope of this project?
Should we do encryption/decryption with a custom aptomictl wrapper script thing?

**Comment by ralekseenkov** [comment]
Thanks for submitting an issue! Secrets is a good topic for discussion. It's definitely in scope of Aptomi.

1) If you go for the approach of having a wrapper, you will need to put decrypted values in dependency YAML before feeding it into aptomictl. It means that Aptomi will store the decrypted values in its database. It will definitely work, but it's not ideal.

2) Having Aptomi shell out to Helm, so that you can use its plugins (e.g. helm-secrets to handle secrets outside of Aptomi) is possible but has some drawbacks. E.g. error handling and distinguishing between different kinds of errors becomes much more difficult when you call Helm as an external binary. But it's something we can discuss.

3) We are planning to have a secrets implementation native to Aptomi. The idea is that secrets will be external and Aptomi will have a plugin to load them from an external source (for a given user, for a given environment, etc).

We actually had a prototype, but decided not to go that route until we come up with a more robust implementation. There is a plugin to load user-specific secrets from "SecretsDir" (encryption is easy to add) is and then reference them from your service:
https://github.com/Aptomi/aptomi/blob/4aba1ecddd160ec67d95161179c7e0d913a95d47/examples/twitter-analytics/policy/John/twitter_stats.yaml#L28-L31

So, all in all, I think:
- wrapper over aptomictl could be a short-term solution
- we could work together to discuss #2 and #3 and see what makes sense

Might also be faster to discuss if you want to join our Slack - http://slack.aptomi.io/

**Comment by Frostman** [comment]
Yeah, I think short term - just use aptomictl wrapper to decrypt sensitive data, but it should be easy to support  decrypting values on aptomi side.
